src/main/java/com/puppycrawl/tools/checkstyle/checks/design/FinalClassCheck.java

```diff
@@ -216,14 +250,11 @@ private void visitClass(DetailAST ast) {
     private void visitCtor(DetailAST ast) {
         if (!ScopeUtil.isInEnumBlock(ast) && !ScopeUtil.isInRecordBlock(ast)) {
             final DetailAST modifiers = ast.findFirstToken(TokenTypes.MODIFIERS);
-            // Can be only of type ClassDesc, preceding if statements guarantee it.
-            final ClassDesc desc = (ClassDesc) typeDeclarations.getFirst();
             if (modifiers.findFirstToken(TokenTypes.LITERAL_PRIVATE) == null) {
+                // Can be only of type ClassDesc, preceding if statements guarantee it.
+                final ClassDesc desc = (ClassDesc) typeDeclarations.getFirst();
                 desc.registerNonPrivateCtor();
```

Comment:
This should be the last code question/change. The code is a lot more understandable now.

Why did we remove `privateCtor` and leave ourselves with `nonPrivateCtor`? This make the other section read as `NOT non-private ctor`, which is a double negative.

Is there some reason we couldn't remove `nonPrivateCtor` and keep `privateCtor` so the other section reads as just a normal `private ctor` check? Let me know if there is something I am missing as I am not expecting other checks or change in regression.

Reply:
@rnveach we can remove the `nonPrivateCtor`  but the test case like 
```
class Check {
    private class Non {
        public Non() {}
        private Non(int i) {}
    }
}
```
will fail. 
To update this we need some additional changes to remove the `nonPrivateCtor` so it requires regression and regression already has heavy changes so it might lead to a bug can we do this in a separate issue?

Reply:
If we can't swap `NOT non-private ctor` with `private ctor` then the name of `private ctor` is wrong/incomplete and there is more to it. I would expect swapping `!!A` with `A` to not require any additional changes.